src/core/Microsoft.PowerToys.Settings.UI/ViewModels/PowerLauncherViewModel.cs

```diff
@@ -1,13 +1,178 @@
-﻿using System;
-
-using Microsoft.PowerToys.Settings.UI.Helpers;
-
-namespace Microsoft.PowerToys.Settings.UI.ViewModels
-{
-    public class PowerLauncherViewModel : Observable
-    {
-        public PowerLauncherViewModel()
-        {
-        }
-    }
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+
+using Microsoft.PowerToys.Settings.UI.Helpers;
+using Microsoft.PowerToys.Settings.UI.Lib;
+using Microsoft.PowerToys.Settings.UI.Views;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.PowerToys.Settings.UI.ViewModels
+{
+    public class PowerLauncherViewModel : Observable
+    {
+        public PowerLauncherSettings settings;
+        private const string POWERTOY_NAME = "PowerLauncher";
+
+
+        public PowerLauncherViewModel()
+        {
+            if (SettingsUtils.SettingsExists(POWERTOY_NAME))
+            {
+                settings = SettingsUtils.GetSettings<PowerLauncherSettings>(POWERTOY_NAME);
+            } else
+            {
+                settings = new PowerLauncherSettings();
+            }
+        }
+
+        private void UpdateSettings([CallerMemberName] string propertyName = null)
+        {
+            // Notify UI of property change
+            OnPropertyChanged(propertyName);
+
+            // Save settings to file
+            var options = new JsonSerializerOptions
+            {
+                WriteIndented = true
+            };
+            SettingsUtils.SaveSettings(JsonSerializer.Serialize(settings, options), POWERTOY_NAME);
+
+            // Propagate changes to Power Launcher through IPC

```

Comment:
lets remove dead code.

Reply:
Actually I just noticed what it does, don't remove it. Let's leave it uncommented out.

Reply:
I uncommented it now.